tokio/src/sync/mutex.rs

```diff
@@ -112,8 +114,12 @@ pub struct Mutex<T> {
 ///
 /// The lock is automatically released whenever the guard is dropped, at which
 /// point `lock` will succeed yet again.
+#[must_use = "if unused the Mutex will immediately unlock"]
```

Comment:
This is not a large enough change for me to care all that much, but since adding the `#[must_use]` attribute isn't directly related to adding `MutexGuard::map`, I'd _kind of_ prefer to have it in a separate PR.

Reply:
This was added when I was merging `MappedMutexGuard` with `MutexGuard`. The `parking_lot` crate has `#[must_use]` on all of its guard types and as a result the original version of `MappedMutexGuard` in this PR had it as well.

It seems like the correct thing to do, and like you said it's a small change, so maybe we should leave it? I can remove it if you want me to.

Reply:
This was added when I was merging `MappedMutexGuard` with `MutexGuard`. The `parking_lot` crate has `#[must_use]` on all of its guard types and as a result the original version of `MappedMutexGuard` in this PR had it as well.

It seems like the correct thing to do, and like you said it's a small change, so maybe we should leave it? I can remove it if you want me to.

Reply:
I have a slight preference for merging it separately, but it's small enough that I don't mind. I definitely agree that we should have it.